### PR TITLE
feat(connector): [Adyen] implement Atome for Adyen

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Atome is a popular buy now, pay later service in some countries of the APAC region.
Line Items must be passed as a meta data 

```
{
  "payment_method": "pay_later",
  "payment_method_type": "atome",
  "payment_method_data": {
    "pay_later": {
      "atome_redirect": {}
    }
  },
  "billing": {
    "address": {
      "line1": "1467",
      "line2": "Harrison Street",
      "line3": "Harrison Street",
      "city": "San Fransico",
      "state": "California",
      "zip": "94122",
      "country": "NO",
      "first_name": "joseph",
      "last_name": "Doe"
    },
    "phone": {
      "number": "8056594427",
      "country_code": "+91"
    }
  },
  "metadata": {
    "order_details": {
      "product_name": "shoe",
      "quantity": 1,
      "amount": 10000
    }
 }
}
```

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/types.rs [hunk 1]
```diff
@@ -827,7 +827,9 @@ impl RequestBody {
         T: std::fmt::Debug,
     {
         router_env::logger::info!(connector_request_body=?body);
-        Ok(Self(Secret::new(encoder(body)?)))
+        let encode = encoder(body)?;
+        crate::logger::debug!("aaaaaaaaaaa{:?}", encode);
```
Comment (prasunna09) [inline crates/router/src/types.rs:831]: Please remove this logger!

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by prasunna09 — APPROVED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

File: config/config.example.toml [hunk 2]
```diff
@@ -319,6 +319,13 @@ payout_eligibility = true   # Defaults the eligibility of a payout method to tru
...
 [pm_filters.adyen]
 online_banking_fpx = {country = "MY", currency = "MYR"}
+online_banking_thailand = {country = "TH", currency = "THB"}
+touch_n_go = {country = "MY", currency = "MYR"}
+atome = {country = "MY,SG", currency = "MYR,SGD"}
 
 [bank_config.online_banking_fpx]
-adyen = { banks = "affin_bank,agro_bank,alliance_bank,am_bank,bank_islam,bank_muamalat,bank_rakyat,bank_simpanan_nasional,cimb_bank,hong_leong_bank,hsbc_bank,kuwait_finance_house,may_bank,ocbc_bank,public_bank,rhb_bank,standard_chartered_bank,uob_bank"}
\ No newline at end of file
+adyen = { banks = "affin_bank,agro_bank,alliance_bank,am_bank,bank_islam,bank_muamalat,bank_rakyat,bank_simpanan_nasional,cimb_bank,hong_leong_bank,hsbc_bank,kuwait_finance_house,may_bank,ocbc_bank,public_bank,rhb_bank,standard_chartered_bank,uob_bank"}
```
Comment (SanchithHegde) [inline config/config.example.toml:327]: Just a suggestion that you can incorporate in further PRs:

You can simplify this as `adyen.banks = "..."`, does not alter the semantics or structure.

Like so:

```suggestion
adyen.banks = "affin_bank,agro_bank,alliance_bank,am_bank,bank_islam,bank_muamalat,bank_rakyat,bank_simpanan_nasional,cimb_bank,hong_leong_bank,hsbc_bank,kuwait_finance_house,may_bank,ocbc_bank,public_bank,rhb_bank,standard_chartered_bank,uob_bank"
```

**Review by ashokkjag — APPROVED** [review]

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 3]
```diff
@@ -749,3 +781,17 @@ fn should_make_adyen_dana_payment_test() {
...
+    tester!(should_make_adyen_online_banking_thailand_payment);
+}
+
+#[test]
+#[serial]
+fn should_make_adyen_touch_n_go_payment_test() {
+    tester!(should_make_adyen_touch_n_go_payment);
+}
+
+// https://hs-payments-test.netlify.app/paypal-redirect?amount=70.00&country=US&currency=USD&mandate_data[customer_acceptance][acceptance_type]=offline&mandate_data[customer_acceptance][accepted_at]=1963-05-03T04:07:52.723Z&mandate_data[customer_acceptance][online][ip_address]=127.0.0.1&mandate_data[customer_acceptance][online][user_agent]=amet%20irure%20esse&mandate_data[mandate_type][multi_use][amount]=700&mandate_data[mandate_type][multi_use][currency]=USD&apikey=dev_uFpxA0r6jjbVaxHSY3X0BZLL3erDUzvg3i51abwB1Bknu3fdiPxw475DQgnByn1z
```
Comment (SanchithHegde) [inline crates/router/tests/connectors/adyen_uk_ui.rs:797]: Do you need this commented line? I believe this was removed in #1763.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**